src/librbd/CMakeLists.txt

```diff
@@ -107,9 +107,12 @@ target_link_libraries(librbd LINK_PRIVATE
   cls_journal_client 
   common
   pthread
-  udev
   ${CMAKE_DL_LIBS}
   ${EXTRALIBS})
+if(HAVE_UDEV)
```

Comment:
@tchaikov 
To start it should use the define for that:
`./cmake/modules/Findudev.cmake:23:# UDEV_LIBRARIES - List of libraries when using udev.`
But I feel uncomfortable removing it, since it was already there, and I try to change as little functionality as possible when adding FreeBSD/Clang specifics.
Old behaviour should be kept as much as possible for all others.

That said, you are right: librbd add udev.
I'll make a separate PR for that, and rebase this one.